bot/exts/filters/token_remover.py

```diff
@@ -116,6 +124,50 @@ async def take_action(self, msg: Message, found_token: Token) -> None:
 
         self.bot.stats.incr("tokens.removed_tokens")
 
+    async def handle_mfa_token(self, msg: Message) -> None:
+        """
+        Check all messages for a string that matches the mfa token pattern.
+
+        Due to how the mfa tokens work, there is no user information supplied for this token.
+        """
+        # Ignore DMs; can't delete messages in there anyway.
+        if not msg.guild or msg.author.bot:
+            return
+
+        was_valid = False
+        for match in MFA_TOKEN_RE.finditer(msg.content):
+
+            if self.is_maybe_valid_hmac(match.group()):
+                was_valid = True
+                break
+        if not was_valid:
+            return
+
+        token = match.group()
+        # since the token was probably valid, we can now delete the message and ping the moderators.
+        # the user is not informed, given the reasoning for deleting the message.
+        with contextlib.suppress(NotFound):
+            await msg.delete()
+
+        log_message = (
+            f"Deleted mfa token sent by {msg.author} in {msg.channel}: "
+            f"{token[:4]}{'x' * len(token[3:-3])}{token[-3:]}"
```

Comment:
```suggestion
            f"{token[:4]}{'x' * len(token[4:-3])}{token[-3:]}"
```
Just so the number of `x`s matches how many characters are missing :)

Reply:
https://github.com/python-discord/bot/pull/2006/commits/8fc503305cde2bf55a22a389bfdce1f7b01b6903